### PR TITLE
Dion admin card

### PR DESCRIPTION
**Pull request by dionyichia**
### 🦋 Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Created admin card with functionalities, without any styling. Added dynamic routing to different organisation pages.

### 🦧 Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

Home screen of all orgs
Screenshot 2025-10-31 at 12 38 28 AM

Dynamic routing to each org page
Screenshot 2025-10-31 at 12 38 35 AM

## 🐸 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

In search bar: http://localhost:3000/user-groups/home-screen.

Click on UserGroup1

need to directly search, homepage not linked to rest of pages yet. I added dummy rows for UserGroup1 only.

## 🐙 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

1. Change of user-groups to user-type
2. Addition of num_users to user-groups table

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### 🐁 Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @me-liu

### CODE REVIEW

File: app/user-groups/[user_group_id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      <section>
+        <h2>Invites</h2>
+        <ul>
+          {invites.map((inv: Invite) => (
+            <li key={inv.invite_id}>{inv.email}</li>
+          ))}
+        </ul>
+      </section>
+
+      <Link href={"/user-groups/home-screen"}>Back to Organisations</Link>
```
Comment (me-liu) [inline app/user-groups/[user_group_id]/page.tsx]: can you make this organizations with a z haha 
usa 🦅🦅🦅

File: app/user-groups/[user_group_id]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  useEffect(() => {
+    if (!user_group_id) {
+      console.log("No user_group_id!");
+      return;
+    }
+
+    console.log(user_group_id);
+
+    loadData();
+  }, [user_group_id]);
```
Comment (me-liu) [inline app/user-groups/[user_group_id]/page.tsx]: maybe do this 
```
useEffect(() => {
  if (user_group_id) {
    loadData();
  }
}, [user_group_id]);
```

will loadData() only when it's truthy. kind of a nit

File: app/user-groups/[user_group_id]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import { useEffect, useState } from "react";
+import Link from "next/link";
+import { useParams } from "next/navigation";
+import { UUID } from "crypto";
+import { fetchInvites } from "@/api/supabase/queries/invites";
+import {
+  fetchUserGroupById,
+  fetchUserGroupMembers,
+} from "@/api/supabase/queries/user-groups";
+import { Invite, Profile, UserGroup } from "@/types/schema";
```
Comment (me-liu) [inline app/user-groups/[user_group_id]/page.tsx]: remove profile

Comment (dionyichia) [reply app/user-groups/[user_group_id]/page.tsx]: added profile for comment below

File: app/user-groups/[user_group_id]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+export default function UserGroupDetailPage() {
+  const { user_group_id } = useParams();
+  const [userGroup, setUserGroup] = useState<UserGroup | null>(null);
+  const [invites, setInvites] = useState<Invite[]>([]);
+  const [allMembers, setAllMembers] = useState<any[]>([]);
```
[the remark this answers is not in the document]
Comment (dionyichia) [reply app/user-groups/[user_group_id]/page.tsx]: ur rightm removed all members state, was using it before but realised i can just chain the command to categorize data into different roles under loadData()
